test/e2e/network/network_policy.go

```diff
@@ -352,12 +486,552 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+			}
+
+			policy, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policy)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(f, policy)
+
+			By("Creating client-a which should not be able to contact the server-b.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", serviceB, 80)
+			})
+
+			By("Creating client-a which should be able to contact the server.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+		})
+	})
+
+	Context("NetworkPolicy between pods, each containing a server container and client container", func() {
+		BeforeEach(func() {
+			By("Creating a pod with simple server container, that serves on port 80 and 81, and a client container.")
+			podServer, service = createPodWithClientServerAndService(f, f.Namespace, "pod-a", []int{80, 81}, "client-a", "pod-a-server-responded")
+
+			By("Waiting for pod ready", func() {
+				err := f.WaitForPodReady(podServer.Name)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			// podServerLabelSelector holds the value for the podSever's label "pod-base-name".
+			podServerLabelSelector = podServer.ObjectMeta.Labels["pod-base-name"]
+
+			// Create pods, which should be able to communicate with the server on port 80 and 81.
+			By("Testing pods can connect to both ports when no policy is present.")
+			testCanConnect(f, f.Namespace, "client-can-connect-80", service, 80)
+			testCanConnect(f, f.Namespace, "client-can-connect-81", service, 81)
+		})
+
+		AfterEach(func() {
+			cleanupServerPodAndService(f, podServer, service)
+		})
+
+		It("should enforce policies to check ingress and egress policies can be controlled independently based on PodSelector [Feature:NetworkPolicy]", func() {
+			var serviceB *v1.Service
+			var podB *v1.Pod
+
+			// Creating pod-b, with server container and client container,  and service-b.
+			// On receiving client request, if communication is successful, the server-b will return a file with "pod-b-server-responded"
+			podB, serviceB = createPodWithClientServerAndService(f, f.Namespace, "pod-b", []int{80}, "client-b", "pod-b-server-responded")
+			By("Waiting for pod-b ready", func() {
+				err := f.WaitForPodReady(podB.Name)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			// Wait for pod-b with service-b to be ready
+			err := framework.WaitForPodRunningInNamespace(f.ClientSet, podB)
+			Expect(err).NotTo(HaveOccurred())
+
+			defer cleanupServerPodAndService(f, podB, serviceB)
+
+			By("Test to check communication from pod-b to pod-a, before applying policies, which should be successful.")
+			shell := "/bin/sh"
+			opt := "-c"
+
+			// Creating command to check communication from pod-b to pod-a
+			// Before applying policy, this communication should be successful
+			targetA := fmt.Sprintf("%s.%s", service.Name, service.Namespace)
+			cmdB := fmt.Sprintf("wget -T 2 %v -O - | grep pod-a-server-responded && exit 0 || exit 1", targetA)
+
+			stdoutB, err := framework.RunKubectl(
```

Comment:
There are wrappers for this. Look at other tests. (Except that also, I don't think you need the combined client-and-server pod, so you don't need to do `kubectl exec`.)